server/src/main/java/org/opensearch/common/util/MovingAverage.java

```diff
@@ -22,14 +22,38 @@ public class MovingAverage {
     private double average = 0;
 
     public MovingAverage(int windowSize) {
-        if (windowSize <= 0) {
-            throw new IllegalArgumentException("window size must be greater than zero");
-        }
-
+        checkWindowSize(windowSize);
         this.windowSize = windowSize;
         this.observations = new long[windowSize];
     }
 
+    public MovingAverage(int newWindowSize, MovingAverage oldMovingAverage) {
```

Comment:
Kind of a style nitpick, but to avoid adding another constructor, I'd make this an instance method so it would look something like:

```
MovingAverage newMovingAverage = oldMovingAverage.copyWithNewWindowSize(newWindowSize);
```

I think the behavior is more clear since it has a method with a meaningful name, and you avoid another constructor.